spec/unit/part_spec.rb

```diff
@@ -30,6 +30,38 @@
       end
     end
 
+    describe '#new' do
+      let(:renderer) do
+        Dry::View::Renderer.new(
+          [Dry::View::Path.new(SPEC_ROOT.join('fixtures/templates'))],
+          format: 'html'
+        )
+      end
+
+      let(:part) { described_class.new(name: name, value: value, context: context, renderer: renderer) }
+
+      context 'same renderer' do
+        it 'renders correctly' do
+          new_part = part.new(value: 'new value')
+          expect(part._render(:hello)).to eql(new_part._render(:hello))
+        end
+      end
+
+      context 'new renderer' do
+        let(:new_renderer) do
+          Dry::View::Renderer.new(
+            [Dry::View::Path.new(SPEC_ROOT.join('fixtures/templates_override'))],
+            format: 'html'
+          )
+        end
+
+        it 'renders correctly' do
+          new_part = part.new(value: 'new value', renderer: new_renderer)
+          expect(part._render(:hello)).to_not eql(new_part._render(:hello))
```

Comment:
how about checking that it renders as expected, instead of checking if it's different than the original one?

Reply:
Sure no problem.